docs/content/migration.md

```diff
@@ -0,0 +1,545 @@
+|--------------------------------------|-----------------------------------------------------------|
+| [`$mdPanel`][109]                    | [`Overlay`][110]                                          |
+| [`md-virtual-repeat`][68]            | [`*cdkVirtualFor`][40]                                    |
+| [`md-virtual-repeat-container`][179] | [`cdk-virtual-scroll-viewport`][40]                       |
+| [`$mdLiveAnnouncer`][111]            | [`LiveAnnouncer`][112]                                    |
+| `$mdUtil`'s [`bidi()`][113] function | [`Directionality` service][114] and [`Dir` directive][180]|
+| [`md-input`][52]'s `md-no-autogrow` and `max-rows` | [`CdkTextareaAutosize`][115]                |
+| [layout][108] system                 | Covered in [separate section](#cdk-layout) of this guide  |
+
+### <a name="theming"></a> Theming
+
+AngularJS Material performs most theming at _run-time_. While this allows developers to configure
+themes dynamically in JavaScript, it also incurs substantial performance cost to generate
+CSS during application load.
+
+Angular Material performs theming at _compile-time_ with [Sass][170]. While this approach
+requires more up-front set-up, it completely removes all JavaScript execution cost associated
+with theming. The use of Sass also moves the theming system to a more well-supported and
+broadly-understood toolchain.
+
+#### Defining palettes
+
+AngularJS Material supports defining [custom palettes][151] or [extending existing palettes][158].
+
+In Angular Material, custom palette creation, including definition of hues and contrast colors, can
+be accomplished by copying one of the [provided palettes][167], modifying it, and then using it in
+your [custom theme creation][164].
+
+#### Defining themes
+
+AngularJS Material uses a provider, [`$mdThemingProvider`][144], to define the theme configuration
+used to generate styles during application loading. This provider also includes a default theme for
+cases when no custom palettes are provided.
+
+Angular Material has no default theme; a theme CSS file must _always_ be included. You can either
+use one of the [pre-built theme files][163], or define a [custom theme][164] with Sass mixins,
+including the output in your application just like any other CSS asset.
+
+##### Defining themes at run-time
+
+AngularJS Material uses a service, [`$mdTheming`][143], to define themes and generate CSS styles at
+run-time.
+
+Defining and generating theme CSS at run-time is not supported by Angular Material. However,
+loading additional theme CSS into your application (or modifying an existing theme's `link` node
+to [swap theme CSS files][181]) at run-time is supported.
+
+Your application can choose to generate additional themes at compile-time or build an API to
+dynamically generate theme CSS on-demand. In either case, it is possible to [lazy load][165] the
+theme CSS at run-time.
```

Comment:
There aren't any official Angular CLI docs (reported via https://github.com/angular/angular/issues/29822) that cover doing this but I found [this StackOverflow answer](https://stackoverflow.com/a/54855112/633107) the provides some, but not all, of the guidance needed.